packages/code-studio/src/storage/LocalWorkspaceStorage.ts

```diff
@@ -21,35 +22,93 @@ const log = Log.module('LocalWorkspaceStorage');
 export class LocalWorkspaceStorage implements WorkspaceStorage {
   static readonly STORAGE_KEY = 'deephaven.WorkspaceStorage';
 
-  static async makeDefaultWorkspaceData(
+  static getBooleanServerConfig(
+    serverConfigValues: Map<string, string> | undefined,
+    key: string
+  ): boolean | undefined {
+    if (serverConfigValues?.get(key)?.toLowerCase() === 'true') {
+      return true;
+    }
+    if (serverConfigValues?.get(key)?.toLowerCase() === 'false') {
+      return false;
+    }
+    return undefined;
+  }
+
+  static makeDefaultWorkspaceSettings(): WorkspaceSettings {
+    return {
+      defaultDateTimeFormat:
+        DateTimeColumnFormatter.DEFAULT_DATETIME_FORMAT_STRING,
+      formatter: [],
+      timeZone: DateTimeColumnFormatter.DEFAULT_TIME_ZONE_ID,
+      showTimeZone: false,
+      showTSeparator: true,
+      disableMoveConfirmation: false,
+      defaultDecimalFormatOptions: {
+        defaultFormatString: DecimalColumnFormatter.DEFAULT_FORMAT_STRING,
+      },
+      defaultIntegerFormatOptions: {
+        defaultFormatString: IntegerColumnFormatter.DEFAULT_FORMAT_STRING,
+      },
+      truncateNumbersWithPound: false,
+      defaultNotebookSettings: {
+        isMinimapEnabled: false,
+      },
+    };
+  }
+
+  static async makeWorkspaceData(
```

Comment:
We're still doing this incorrectly - this `LocalWorkspaceStorage.makeWorkspaceData` should _not_ take in the `serverConfigValues`. Just provide an empty object for `settings` here (e.g. no default overrides).
The `makeDefaultWorkspaceSettings` method above _should_ take the serverConfigValues, and populate the settings there if the settings if the server overrides the settings. Whatever the server provides are the "defaults".

Reply:
Yeah think I was trying to keep our default and put the server config defaults along with the user set values. What I should've done is override our defaults with server config defaults, and then keep the user set values separately.